src/singletons/Settings.hpp

```diff
@@ -168,6 +168,7 @@ class Settings
     FloatSetting boldScale = {"/appearance/boldScale", 63};
     BoolSetting showTabCloseButton = {"/appearance/showTabCloseButton", true};
     BoolSetting showTabLive = {"/appearance/showTabLiveButton", true};
+    BoolSetting compactTabs = {"/appearance/compactTabs", false};
```

Comment:
Can you make the this an `EnumSetting<TabStyle>` instead?

Define `TabStyle` as
```c++
enum class TabStyle : std::uint8_t {
  Normal,
  Compact,
};
```

In the general page, use `SettingWidget` instead

```c++
SettingWidget::dropdown("Tab style", s.tabStyle)
  ->addTo(layout);
```


With this change, you might want to implement a few anonymous functions inside `NotebookTab.cpp` to get the relevant value based on the tab style.
Something like
```c++
float getCompactDivider(TabStyle tabStyle) {
  switch (tabStyle) {
    case TabStyle::Normal:
      return 1.5;
    case TabStyle::Compact:
      return 1.0;
  }
}
```

and same for the `compactReducer`

Reply:
An option to have the tabs as wide as normal but with the tab close button hidden would be nice too. In the future, an option to stretch the tabs to the width of the window would be great.

Reply:
Yeah, I feel like there's more styles than just "Compact" that we might want to implement. touch-friendly tabs!